src/NServiceBus.Gateway/Deduplication/LegacyDeduplicationWrapper.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus.Gateway
+{
+    using System;
+    using System.Threading.Tasks;
+    using Deduplication;
+    using Extensibility;
+
+    class LegacyDeduplicationWrapper : IGatewayDeduplicationStorage
```

Comment:
Why not just drop support for the old interface? It gets moved to the gateway project anyway and is a breaking change which means recompiling seams.

Reply:
the plan is to release a new minor of the gateway, so that's not the time for breaking changes.

Reply:
My bad @timbussmann I thought this `IGatewayDeduplicationStorage` was cloned to the gateway but this is the interface defined in the core assembly.